test/functionality/r-bridge/lang/ast/parse-access.ts

```diff
@@ -342,157 +342,409 @@ describe('Parse value access', withShell(shell => {
-				}
-			},null,null]
-		}))
+						type:     RType.Argument,
+						location: rangeFrom(1, 4, 1, 4),
+						lexeme:   '5',
+						name:     undefined,
+						info:     {},
+						value:    {
+							type:     RType.Number,
+							location: rangeFrom(1, 4, 1, 4),
+							lexeme:   '5',
+							content:  numVal(5),
+							info:     {}
+						}
+					}, null, null]
+				})
+			},
+			{
+				step:   DESUGAR_NORMALIZE,
+				wanted: exprList({
+					type:         RType.FunctionCall,
+					location:     rangeFrom(1, 2, 1, 3),
+					lexeme:       '[[',
+					info:         {},
+					functionName: {
+						type:      RType.Symbol,
+						lexeme:    '[[',
+						content:   '[[',
+						info:      {},
+						location:  rangeFrom(1, 2, 1, 3),
+						namespace: undefined
+					},
+					flavor:    'named',
+					arguments: [{
+						type:      RType.Symbol,
+						location:  rangeFrom(1, 1, 1, 1),
+						namespace: undefined,
+						lexeme:    'b',
+						content:   'b',
+						info:      {}
+					}, {
+						type:     RType.Argument,
+						location: rangeFrom(1, 4, 1, 4),
+						lexeme:   '5',
+						name:     undefined,
+						info:     {},
+						value:    {
+							type:     RType.Number,
+							location: rangeFrom(1, 4, 1, 4),
+							lexeme:   '5',
+							content:  numVal(5),
+							info:     {}
+						}
+					}, undefined, undefined]
+				})
+			}
+		])
 	})
 	describe('Dollar and Slot', () => {
-		assertAst('Dollar access', shell, 'c$x', exprList({
-			type:     RType.Access,
-			location: rangeFrom(1, 2, 1, 2),
-			lexeme:   '$',
-			operator: '$',
-			info:     {},
-			accessed: {
-				type:      RType.Symbol,
-				location:  rangeFrom(1, 1, 1, 1),
-				namespace: undefined,
-				lexeme:    'c',
-				content:   'c',
-				info:      {}
+		assertAst('Dollar access', shell, 'c$x', [
+			{
+				step:   NORMALIZE,
+				wanted: exprList({
+					type:     RType.Access,
+					location: rangeFrom(1, 2, 1, 2),
+					lexeme:   '$',
+					operator: '$',
+					info:     {},
+					accessed: {
+						type:      RType.Symbol,
+						location:  rangeFrom(1, 1, 1, 1),
+						namespace: undefined,
+						lexeme:    'c',
+						content:   'c',
+						info:      {}
+					},
+					access: 'x'
+				})
 			},
-			access: 'x'
-		}))
-		assertAst('Slot based access', shell, 'd@y', exprList({
-			type:     RType.Access,
-			location: rangeFrom(1, 2, 1, 2),
-			lexeme:   '@',
-			operator: '@',
-			info:     {},
-			accessed: {
-				type:      RType.Symbol,
-				location:  rangeFrom(1, 1, 1, 1),
-				namespace: undefined,
-				lexeme:    'd',
-				content:   'd',
-				info:      {}
+			{
+				step:   DESUGAR_NORMALIZE,
+				wanted: exprList({
+					type:         RType.FunctionCall,
+					location:     rangeFrom(1, 2, 1, 2),
+					lexeme:       '$',
+					info:         {},
+					functionName: {
+						type:      RType.Symbol,
+						lexeme:    '$',
+						content:   '$',
+						info:      {},
+						location:  rangeFrom(1, 2, 1, 2),
+						namespace: undefined
```

Comment:
This should be wrong! it should be a `InternalScope` (`".$internal"`). Please sync again with `572-normalize-v2-desugaring` and check if the behavior is now as intended. (The reason is that `$` and `@` work different wrt. hiding them with a function definition).

Reply:
maybe this time i can even rebase correctly who knows